src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.cs

```diff
@@ -43,5 +43,68 @@ public sealed override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationAction(CompilationAnalyzer.AnalyzeCompilation);
         }
+
+        protected bool IsNotSupportedByLiveAnalysis(int code)
+        {
+            if (code == MessageProvider.ERR_FailedToCreateTempFile ||
```

Comment:
Btw I meant this only for diagnostics from CompilerAnalyzer: "Live analysis is only supported for diagnostics with location in source".